linkerd/install.go

```diff
@@ -118,7 +118,15 @@ func (iClient *Client) downloadFile(urlToDownload, localFile string) error {
 		logrus.Error(err)
 		return err
 	}
-	defer dFile.Close()
+
+	defer dFile.Close() // #nosec
```

Comment:
I am willing to accept the tradeoff of more readable and easily understandable code for needing to obsessively review code to ensure that the file is closed in all cases, and that’s the approach I recommend in code reviews I give to others.

Reply:
@Aisuko, can we create a callFuncAndLogError function that takes function as a parameter and execute the function and if error occurred, just log it, we can then always defer this function by passing the function that needs to be deferred as a parameter...

Reply:
@nveenjain Good point. I know what's your mean, the ` func` as a parameter. I'd like to try the solution we discussed. Maybe after the Github action Issue

Reply:
Gents, I'm withholding opinion here. Given your discussion, shall we merge this and circle back with a GitHub Action?